test/dapp.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { BrowserContext, Page } from 'playwright-core';
+import { Dappwright } from '../src/index';
+import launchBrowser from './helpers/launchBrowser';
+
+describe('when interacting with dapps', () => {
+  let browserContext: BrowserContext, wallet: Dappwright, testPage: Page;
+
+  beforeAll(async () => {
+    [browserContext, testPage, wallet] = await launchBrowser();
+    await wallet.switchNetwork('local');
+    await testPage.bringToFront();
+  });
+
+  afterAll(async () => {
+    await browserContext.close();
+  });
+
+  it('should be able to connect', async () => {
+    await testPage.click('.connect-button');
+    await wallet.approve();
+
+    await testPage.waitForSelector('#connected');
+  });
+
+  it('should be able to sign messages', async () => {
+    await testPage.click('.sign-button');
+    await wallet.sign();
+
+    await testPage.waitForSelector('#signed');
+  });
+
+  describe('when confirming a transaction', () => {
+    it('should be able to confirm without altering gas settings', async () => {
+      await testPage.click('.increase-button');
+      await wallet.confirmTransaction();
+
+      // confirms transasction sent successfully
+      await testPage.waitForSelector('#increased');
+    });
+
+    it('should be able to confirm with custom gas settings', async () => {
```

Comment:
should there be another test that uses the `increase-fees` button?

Reply:
that's dormant functionality I think. Will create an issue to investigate.

Reply:
[]()